nixos/modules/services/networking/tinc.nix

```diff
@@ -163,12 +163,7 @@ in
         wantedBy = [ "multi-user.target" ];
         after = [ "network.target" ];
         path = [ data.package ];
-        restartTriggers =
-          let
-            drvlist = [ config.environment.etc."tinc/${network}/tinc.conf".source ]
-                        ++ mapAttrsToList (host: _: config.environment.etc."tinc/${network}/hosts/${host}".source) data.hosts;
-          in # drvlist might be too long to be used directly
-            [ (builtins.hashString "sha256" (concatMapStrings (d: d.outPath) drvlist)) ];
```

Comment:
Does it not forget keys once no peer has it longer in its list?

Reply:
It can get keys from another peer via gossip-protocol.
Do you remember we added support for ```tinc.${netname}``` users?
Before that, ```/etc/tinc/${network}/hosts``` was owned by root and thus read-only for the daemon.
Now the ```tinc``` daemon is able to create new files in hosts based on gossip info.

I have no idea what would be the best way to clean up the directory from obsolete keys (may be by creating empty root-owned files?)